pkg/edgeview/src/system.go

```diff
@@ -1306,6 +1315,140 @@ func runTechSupport(cmds cmdOpt, isLocal bool) {
 	}
 }
 
+// edgeview request debug container to generate collect-info and
+// later to remove the tar.gz file
+// the protocol is for edgeview to create a file
+// /run/edgeview/edgeview-request-collect-info, and debug container
+// run 'collect-info.sh' to generate the tar.gz file, when the job is
+// done, remove the file /run/edgeview/edgeview-request-collect-info.
+// edgeview can also request to remove the tar.gz file by creating
+// /run/edgeview/edgeview-request-remove-tar.gz file, and debug container
+// will remove the generated tar.gz file.
+func runCollectInfo() {
+	if _, err := os.Stat(requestCollectInfoFile); err == nil {
+		fmt.Printf("runCollectInfo, collect info request already in progress\n")
+		return
+	}
+
+	startTime := time.Now() // Record the start time
+
+	err := touchRequestFile(requestCollectInfoFile, "")
+	if err != nil {
+		fmt.Printf("runCollectInfo, can not create collect info request file\n")
+		return
+	}
```

Comment:
can i do a rename() operation in this.

Reply:
You're doing now the rename, but still it is not atomic, because first you do `os.Stat` and then you call `touchRequestFile`.

Reply:
Hmm, the os.Stat() is only error condition check, it jumps out in case there is such a file exist, it has nothing to do with the newly touched file being generated. Do i miss something?

Reply:
Isn't the `os.Stat` there to check that no other collect-info is currently running?

What happens if the user starts two at the same time? Then for both the error return value of `os.Stat` will be not nil and they both continue.

Reply:
this will be extremely unlikely event. even it happens, not much problem results: yes, they both continue, which means the same request file is generated and regenerated. the debug task is working on this, and announce the tarfile is done. say both edgeview sessions will download the tarfile. The fast one will finish first and announce we can remove the tarfile, the second one will get an error during still downloading, saying copy file failed. The 2nd guy just have to retry one more time.

Reply:
If you don't want to fix it, I think this text should be added to the code as a comment.

Reply:
Thx. Resolved.